data/gpx_test_data/color.gpx

```diff
@@ -0,0 +1,80 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<gpx xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.topografix.com/GPX/1/1" xsi:schemaLocation="http://www.topografix.com/GPX/1/1 http://www.topografix.com/GPX/1/1/gpx.xsd http://www.garmin.com/xmlschemas/GpxExtensions/v3 http://www.garmin.com/xmlschemas/GpxExtensionsv3.xsd http://www.garmin.com/xmlschemas/TrackPointExtension/v1 http://www.garmin.com/xmlschemas/TrackPointExtensionv1.xsd http://www.topografix.com/GPX/gpx_style/0/2 http://www.topografix.com/GPX/gpx_style/0/2/gpx_style.xsd" xmlns:gpxtpx="http://www.garmin.com/xmlschemas/TrackPointExtension/v1" xmlns:gpxx="http://www.garmin.com/xmlschemas/GpxExtensions/v3" xmlns:gpx_style="http://www.topografix.com/GPX/gpx_style/0/2" version="1.1" creator="https://gpx.studio">
+<metadata>
+    <name>new</name>
+    <author>
+        <name>gpx.studio</name>
```

Comment:
By some reason I can't install this app on my android phone, could you please share a recorded file here (all points could be removed for privacy, I'm just interested in metadata and track data).

Reply:
Looks like it was a file from another logger: https://gpslogger.app/
It has only points in the log, without any metadata, with a file name like `20230402.gpx`:

```
<?xml version="1.0" encoding="UTF-8" ?><gpx version="1.0" creator="GPSLogger 126 - http://gpslogger.mendhak.com/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.topografix.com/GPX/1/0" xsi:schemaLocation="http://www.topografix.com/GPX/1/0 http://www.topografix.com/GPX/1/0/gpx.xsd"><time>2023-04-02T11:25:10.964Z</time><trk><trkseg><trkpt lat="...
```

Reply:
> Looks like it was a file from another logger: https://gpslogger.app/ It has only points in the log, without any metadata, with a file name like `20230402.gpx`:
> 
> ```
> <?xml version="1.0" encoding="UTF-8" ?><gpx version="1.0" creator="GPSLogger 126 - http://gpslogger.mendhak.com/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.topografix.com/GPX/1/0" xsi:schemaLocation="http://www.topografix.com/GPX/1/0 http://www.topografix.com/GPX/1/0/gpx.xsd"><time>2023-04-02T11:25:10.964Z</time><trk><trkseg><trkpt lat="...
> ```
Ok, I see.
So we want to add filename as a parameter to
`std::unique_ptr<kml::FileData> LoadKmlData(Reader const & reader, KmlFileType fileType)`
method so it can pass it to GPX deserializer and it will fallback to it in case of missing name in file?

Reply:
That can be an option, right.

Alternatively, is it better/easier to check if returned FileData has empty Category name, and fix it there?

Reply:
Yes, thanks, it looks easier. I applied this change and added a test for this case.